### ISSUE TITLE
(/bin/sh^M: bad interpreter) Script Folder, Line ending in DOS Windows (CRLF) will confuse many UNIX bash interpreters

### ISSUE THREAD

**Issue by JOELwindows7** [reticulatedpines/qemu-eos#3]
# According to this message at [here](https://www.magiclantern.fm/forum/index.php?topic=20214.msg241006#msg241006 )
e.g. with the `scripts/git-submodule.sh` used during building will have this weird error
```
joelwindows7@JOEL-ROG-GL503GE:~/sauce/simplified/qemu-eos-build$ ./scripts/git-submodule.sh
-bash: ./scripts/git-submodule.sh: /bin/sh^M: bad interpreter: No such file or directory
```
which basically the bash (like in my WSL) confused and thinks you had `/bin/sh^M` at the first line, while actually not.

This is due to a fact I just figured out that the line ending format is in DOS (`CRLF`) which happens to confuse various Bash interpreters including mine.

Now what I did to fix this problem is to convert every file possible inside the `qemu-eos/script` from DOS Windows (`CRLF`) to UNIX line ended (`LF`). So I went to that script folder & do this command:
```
for FILER in ./*; do chmod +x $FILER; dos2unix $FILER;
```
Basically, for each file, add executable right & also convert the line format from CRLF to LF. Forget the former since permission won't uploads into git. the latter does.

I suggest that should also convert your script and maybe others to such line ending format & push them all here so others will not encounter such error. No idea if others don't have this error, but just in case, it's recommended to do so anyways.

See [this answer from Ask Ubuntu](https://askubuntu.com/a/861053/64076) on the `dos2unix` utility.

**Comment by reticulatedpines** [comment]
That file doesn't contain any CR characters (^M).  This is probably your local system configuration, most likely for git, inserting carriage return characters.  Here's the expected sha1sum for the file:

```
$ sha1sum ../../scripts/git-submodule.sh 
8c3fb0a4b22a5b31e1d68083c108db149192bb37  ../../scripts/git-submodule.sh

```
This is configurable, I recommend making git never change line endings (modern Windows systems can handle unix-style line endings).

https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

The files are also marked executable here on my Linux system, I suspect this is also either git config, or a WSL problem.
